torch/csrc/jit/tracer.h

```diff
@@ -49,9 +49,11 @@ TORCH_API void setRecordSourceLocation(void (*v)(Node*));
 // involving this variable know which node in the IR to reference.
 TORCH_API void setValueTrace(const IValue& v, Value* value);
 
+TORCH_API void setFutureTrace(const c10::intrusive_ptr<c10::ivalue::Future>& fut, Value* value);
+
 inline void delValueTrace(const Variable& var) {
   AT_ASSERT(var.defined());
-  getTracingState()->value_map.erase(var);
+  getTracingState()->env_stack.back().value_map.erase(var);
```

Comment:
This does not walk the stack. This will fail if a in-place resize_ is used inside a fork block.